Gems/AWSCore/Code/Source/Editor/Attribution/AWSCoreAttributionManager.cpp

```diff
@@ -71,7 +69,8 @@ namespace AWSCore
         AZStd::string editorAWSPreferencesFilePath =
             AZStd::string::format("@user@/%s/%s", AZ::SettingsRegistryInterface::RegistryFolder, EditorAWSPreferencesFileName);
         AZStd::array<char, AZ::IO::MaxPathLength> resolvedPathAWSPreference{};
-        if (!fileIO->ResolvePath(editorAWSPreferencesFilePath.c_str(), resolvedPathAWSPreference.data(), resolvedPathAWSPreference.size()))
+        if (!fileIO->ResolvePath(
+                editorAWSPreferencesFilePath.c_str(), resolvedPathAWSPreference.data(), AZ_ARRAY_SIZE(resolvedPathAWSPreference)))
```

Comment:
There s a simpler signature for this that uses an AZ::IO::FixedMaxPath


```
virtual bool ResolvePath(AZ::IO::FixedMaxPath& resolvedPath, const AZ::IO::PathView& path) const = 0;
```

You can just make a fixed string backed path object and use that to call ResolvePath
```suggestion
        AZ::IO::FixedMaxPath resolvedPathAWSPreference;
        if (!fileIO->ResolvePath(resolvedPathAWSPreference, AZ::IO::PathView(editorAWSPreferencesFilePath)))
```